test/project_types/script/layers/infrastructure/script_project_repository_test.rb

```diff
@@ -353,63 +393,180 @@ def hash_except(config, *keys)
+        file_content = JSON.parse(ctx.read(script_config_filename))
 
-        assert_equal new_title, script_json.title
+        assert_equal new_title, script_config.title
         assert_equal new_title, file_content["title"]
-        refute_equal initial_title, script_json.title
+        refute_equal initial_title, script_config.title
 
-        assert_equal initial_description, script_json.content["description"]
+        assert_equal initial_description, script_config.content["description"]
         assert_equal initial_description, file_content["description"]
-        assert_equal script_json_content["version"], script_json.version
-        assert_equal script_json_content["version"], file_content["version"]
-        assert_equal script_json_content["configuration"].to_json, script_json.configuration.to_json
-        assert_equal script_json_content["configuration"].to_json, file_content["configuration"].to_json
+        assert_equal script_config_content["version"], script_config.version
+        assert_equal script_config_content["version"], file_content["version"]
+        assert_equal script_config_content["configuration"].to_json, script_config.configuration.to_json
+        assert_equal script_config_content["configuration"].to_json, file_content["configuration"].to_json
+      end
+    end
+  end
+
+  describe "ScriptConfigYmlRepository" do
+    let(:instance) { Script::Layers::Infrastructure::ScriptProjectRepository::ScriptConfigYmlRepository.new(ctx: ctx) }
+    let(:version) { "2" }
+    let(:title) { "title" }
+    let(:content) { { "version" => version, "title" => title }.to_yaml }
+
+    describe "get" do
+      subject { instance.get }
+
+      describe "when file does not exist" do
+        it "returns nil" do
+          assert_nil(subject)
+        end
+      end
+
+      describe "when file exists" do
+        before do
+          File.write(script_config_filename, content)
+        end
+
+        describe "when content is invalid yaml" do
+          let(:content) { "*" }
+
+          it "raises InvalidScriptConfigYmlDefinitionError" do
+            assert_raises(Script::Layers::Infrastructure::Errors::InvalidScriptConfigYmlDefinitionError) { subject }
+          end
+        end
+
+        describe "when content is not a hash" do
+          let(:content) { "" }
+
+          it "raises InvalidScriptConfigYmlDefinitionError" do
+            assert_raises(Script::Layers::Infrastructure::Errors::InvalidScriptConfigYmlDefinitionError) { subject }
+          end
+        end
+
+        describe "when content is missing fields" do
+          let(:content) { {}.to_yaml }
+
+          it "raises MissingScriptConfigYmlFieldError" do
+            assert_raises(Script::Layers::Infrastructure::Errors::MissingScriptConfigYmlFieldError) { subject }
+          end
+        end
+
+        describe "when content is valid yaml" do
+          it "returns the entity" do
+            assert_equal version, subject.version
```

Comment:
Should we also check the other script config values here?

Reply:
Done